File: Backend/schemas/lobby.py
from pydantic import BaseModel
from .player import Player

FLOWS: dict = {
        "CLASSIC": ["PROMPT", "RECORDING"],
        "BLIND_KARAOKE": ["RECORDING"] 
    }

class Lobby(BaseModel):
    lobbyId: str
    players: dict[str, Player]  # nickname: Player 
    lobbyHost: str
    gameMode: str = "CLASSIC"
    settings: dict = {}
    gameStarted: bool = False
    roundIndex: int = 1
    recList: list = []
    numRounds: int = 1
    roundNum: int = 1

File: Frontend/src/types/lobby.ts
import type { Player } from "./player.ts";

export interface Lobby {
    // Basic Info
    lobbyId: string;      // maps to room_code in DB
    lobbyHost: string;    // maps to host_nickname
    players: Record<string, Player>;

    // Game Configuration
    gameMode: string;     // maps to game_mode
    settings: Record<string, any>;

    gameStarted: boolean; // maps to game_started
    roundNum: number;     // maps to round_num (1, 2, 3...)
    numRounds: number;    // maps to num_rounds (usually total player count)
    recList: any[][];     // maps to rec_list (The actual audio/note data)
}